### PR TITLE
iotbus: Implement pinmap draft

### PR DESCRIPTION
**Pull request by pillip8282**
Add iotbus pinmap
TizenRT configure ports reading CONFIG_PROD_HEADER which is defined
Kconfig. If the port configuration need to be changed then change
CONFIG_PROD_HEADER in 'Board Selection' > 'Product Pin Definition'
then it'll refer the changed header file
By default it read 'prod/fac.h' so this file has to be in prod folder.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [bb167fbf41babe9d0391711420574053e4df0db3] - Code Rule Check Result:
os/arch/arm/src/imxrt/imxrt_adc.c:91: ERROR: [SPC_M_SEP] space prohibited before that close square bracket ']'
os/arch/arm/src/imxrt/imxrt_adc.h:130: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.h:131: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.h:141: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.h:155: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_gpio.c:770: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/arch/arm/src/imxrt/imxrt_gpio.c:774: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/arch/arm/src/imxrt/imxrt_gpio.c:774: ERROR: [BRC_M_SMT] else should follow close brace '}'
os/arch/arm/src/imxrt/imxrt_gpio.h:385: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/board/imxrt1020-evk/src/imxrt_boot.c:187: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [bb167fbf41babe9d0391711420574053e4df0db3] - Code Rule Check (C++) OK.

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 1]
```diff
@@ -34,21 +34,65 @@
...
+		IMXRT_PADMUX_GPIO_AD_B1_09_INDEX,	/* ADC2 Pin 9 */					// 중복
+		IMXRT_PADMUX_GPIO_AD_B1_10_INDEX,	/* ADC2 Pin 10 */					// 중복
+		IMXRT_PADMUX_GPIO_AD_B1_11_INDEX,	/* ADC2 Pin 11 */					// 중복
+		IMXRT_PADMUX_GPIO_AD_B1_12_INDEX,	/* ADC2 Pin 12 */					// 중복
+		IMXRT_PADMUX_GPIO_AD_B1_13_INDEX,	/* ADC2 Pin 13 */					// 중복
+		IMXRT_PADMUX_GPIO_AD_B1_14_INDEX,	/* ADC2 Pin 14 */					// 중복
+		IMXRT_PADMUX_GPIO_AD_B1_15_INDEX,	/* ADC2 Pin 15 */					// 중복
+};
+
+static FAR const uint8_t *g_adc_padmux[IMXRT_ADC_NBUS ] = {
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: Please remove space before ]

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 2]
```diff
@@ -730,6 +781,7 @@ static const struct adc_ops_s g_adcops = {
 	.ao_ioctl = adc_ioctl,
 };
 
+#if 0
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: Please remove this code if not needed.

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 3]
```diff
@@ -860,4 +912,102 @@ struct adc_dev_s *imxrt_adc_initialize(int bus, int channel)
...
+ ************************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: imxrt_gpio_configinput
+ ****************************************************************************/
+
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: Please remove this extra line.

**Comment by seinfra** [comment]
Target : [3c1b7496005f333f4d997bc3f45351cf40ccbc28] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3c1b7496005f333f4d997bc3f45351cf40ccbc28] - Code Rule Check OK.

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 4]
```diff
@@ -860,4 +912,102 @@ struct adc_dev_s *imxrt_adc_initialize(int bus, int channel)
...
+	FAR const uint8_t *table;
+	iomux_pinset_t ioset;
+	uintptr_t regaddr;
+	unsigned int index;
+
+	DEBUGASSERT((unsigned int)bus < IMXRT_ADC_NBUS);
+
+	/* Configure pin as in input */
+
+//	imxrt_gpio_dirin(port, pin);		// ??
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: Remove if not needed.

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 5]
```diff
@@ -860,4 +912,102 @@ struct adc_dev_s *imxrt_adc_initialize(int bus, int channel)
...
+
+	index = (unsigned int)table[channel];
+	if (index >= IMXRT_PADMUX_NREGISTERS) {
+		return -EINVAL;
+	}
+
+	regaddr = imxrt_padmux_address(index);
+	putreg32(PADMUX_MUXMODE_ALT5, regaddr);
+
+//	imxrt_gpio_select(bus, channel);
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: same as above.

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 6]
```diff
@@ -860,4 +912,102 @@ struct adc_dev_s *imxrt_adc_initialize(int bus, int channel)
...
+	int bus = (pinset & GPIO_PORT_MASK) >> GPIO_PORT_SHIFT;
+	int channel = (pinset & GPIO_PIN_MASK) >> GPIO_PIN_SHIFT;
+
+	if (channel > IMXRT_ADC_NCHANEL) {
+		allvdbg("IMXRT ADC1 has maximum %d ADC channels.\n", IMXRT_ADC_NCHANEL);
+		return NULL;
+	}
+
+	adcdev = (struct adc_dev_s *)
+			kmm_malloc(sizeof(struct adc_dev_s));
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: Please remove spaces or tabs between typecast and kmm_malloc.
like: adcdev = (struct adc_dev_s *)kmm_malloc(sizeof(struct adc_dev_s));

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 7]
```diff
@@ -860,4 +912,102 @@ struct adc_dev_s *imxrt_adc_initialize(int bus, int channel)
...
+	if (!adcdev) {
+		return NULL;
+	}
+	priv = (struct imxrt_dev_s *)
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: same as above.

File: os/arch/arm/src/imxrt/imxrt_adc.h [hunk 8]
```diff
@@ -500,9 +569,13 @@ void imxrt_adc_clearstatusflags(ADC_Type *base, uint32_t mask);
  *   Valid can device structure reference on success; a NULL on failure
  *
  ****************************************************************************/
-struct adc_dev_s *imxrt_adc_initialize(int bus, int cchannels);
+//struct adc_dev_s *imxrt_adc_initialize(int bus, int cchannels);
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.h]: Please remove it if not needed.

File: os/arch/arm/src/imxrt/imxrt_gpio.c [hunk 9]
```diff
@@ -357,10 +433,10 @@ static FAR const uint8_t *g_gpio_padmux[IMXRT_GPIO_NPORTS + 1] = {
 	g_gpio1_padmux,				/* GPIO1 */
 	g_gpio2_padmux,				/* GPIO2 */
 	g_gpio3_padmux,				/* GPIO3 */
-	#if !defined(CONFIG_ARCH_CHIP_FAMILY_IMXRT102x)
+//#if !defined(CONFIG_ARCH_CHIP_FAMILY_IMXRT102x)
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_gpio.c]: same as above.

File: os/arch/arm/src/imxrt/imxrt_adc.c (hunk 1 above)
Comment (pillip8282) [reply os/arch/arm/src/imxrt/imxrt_adc.c]: thanks I'll fix

File: os/arch/arm/src/imxrt/imxrt_adc.c (hunk 2 above)
Comment (pillip8282) [reply os/arch/arm/src/imxrt/imxrt_adc.c]: thanks I'll fix

File: os/arch/arm/src/imxrt/imxrt_adc.c (hunk 3 above)
Comment (pillip8282) [reply os/arch/arm/src/imxrt/imxrt_adc.c]: thanks I'll fix

File: os/arch/arm/src/imxrt/imxrt_adc.c (hunk 4 above)
Comment (pillip8282) [reply os/arch/arm/src/imxrt/imxrt_adc.c]: thanks I'll fix

File: os/arch/arm/src/imxrt/imxrt_adc.c (hunk 5 above)
Comment (pillip8282) [reply os/arch/arm/src/imxrt/imxrt_adc.c]: thanks I'll fix

File: os/arch/arm/src/imxrt/imxrt_adc.c (hunk 6 above)
Comment (pillip8282) [reply os/arch/arm/src/imxrt/imxrt_adc.c]: thanks I'll fix

File: os/arch/arm/src/imxrt/imxrt_adc.c (hunk 7 above)
Comment (pillip8282) [reply os/arch/arm/src/imxrt/imxrt_adc.c]: thanks I'll fix

File: os/arch/arm/src/imxrt/imxrt_adc.h (hunk 8 above)
Comment (pillip8282) [reply os/arch/arm/src/imxrt/imxrt_adc.h]: thanks I'll fix

File: os/arch/arm/src/imxrt/imxrt_gpio.c (hunk 9 above)
Comment (pillip8282) [reply os/arch/arm/src/imxrt/imxrt_gpio.c]: thanks I'll fix

**Comment by seinfra** [comment]
Target : [0fadeccd754e187b4f8f171fb119fafa906b7dfe] - Code Rule Check Result:
os/board/imxrt1020-evk/src/prod/fac.h:10: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1020-evk/src/prod/fac.h:11: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1020-evk/src/prod/fac.h:12: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1020-evk/src/prod/fac.h:13: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1020-evk/src/prod/fac.h:23: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1020-evk/src/prod/fac.h:24: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1020-evk/src/prod/fac.h:25: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1020-evk/src/prod/fac.h:26: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1020-evk/src/prod/fac.h:29: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/board/imxrt1020-evk/src/prod/fac.h:30: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1020-evk/src/prod/fac.h:31: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1020-evk/src/prod/fac.h:32: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1020-evk/src/prod/fac.h:33: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:10: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:11: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:12: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:13: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:23: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:24: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:25: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:26: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:29: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/board/imxrt1050-evk/src/prod/fac.h:30: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:31: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:32: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/imxrt1050-evk/src/prod/fac.h:33: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [0fadeccd754e187b4f8f171fb119fafa906b7dfe] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [24147ab271e27d2b9b73274aafa4ddab6b0f262d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [24147ab271e27d2b9b73274aafa4ddab6b0f262d] - Code Rule Check OK.

**Pull request merged**